packages/orchestration/src/exos/cosmos-orchestration-account.js

```diff
@@ -938,11 +938,12 @@ export const prepareCosmosOrchestrationAccountKit = (
 
             // set a `timeoutTimestamp` if caller does not supply either `timeoutHeight` or `timeoutTimestamp`
             // TODO #9324 what's a reasonable default? currently 5 minutes
-            const timeoutTimestampVowOrValue =
-              opts?.timeoutTimestamp ??
-              (opts?.timeoutHeight
-                ? 0n
-                : E(timestampHelper).getTimeoutTimestampNS());
+            const timeoutTimestampVowOrValue = (() => {
+              if (opts && 'timeoutHeight' in opts) return 0n;
+              if (opts && 'timeoutTimestamp' in opts)
+                return opts.timeoutTimestamp;
+              return E(timestampHelper).getTimeoutTimestampNS();
```

Comment:
I would use the same code as `local-orch-account-.js`  here and make sure we use `opts.timeoutRelativeSeconds`

Reply:
I can see 2 or 3 bugs that I introduced in this area, so I moved this code to a method on `timestampHelper` and added unit tests.